Sources/SpeziOnboarding/OnboardingIdentifiableViewModifier.swift

```diff
@@ -0,0 +1,39 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// Wrap `Content` `View` in an `Identifiable` `View`.
+public struct OnboardingIdentifiableView<Content>: View, Identifiable where Content: View {
+    /// Unique identifier of the wrapped `View`.
+    public let id: String
+    /// Wrapped `View`.
+    public var body: Content
+}
+
+struct OnboardingIdentifiableViewModifier: ViewModifier {
+    let identifier: String
+
+    func body(content: Content) -> some View {
+        OnboardingIdentifiableView(
+            id: self.identifier,
+            body: content
+        )
+    }
+}
+
+
+extension View {
+    /// `ViewModifier` assigning an identifier to the `View` it is applied to.
+    /// When applying this modifier repeatedly, the outermost ``id(_:)`` counts.
```

Comment:
Does it make sene to add a `Note` bubble to the documentation, that this modifier only has an effect if the view modifier is directly used in the OnboardingStack view builder (maybe even demonstrate this with a short code example)?